File: 2021/12/01/thousand_separator.py
"""
https://leetcode.com/problems/thousand-separator/
Author: Andrew Jarombek
Date: 12/1/2021
"""


def thousand_separator(n: int) -> str:
    if n < 1000:
        return f'{n}'

    result = []
    digit = 0

    while n > 0:
        value = n % 10

        if digit % 3 == 0:
            result.append('.')

        result.append(f'{value}')

        n //= 10
        digit += 1

    return "".join(result[:0:-1])

File: 2022/09/01/java/ThousandSeparator.java
/**
 * https://leetcode.com/problems/thousand-separator/
 * @author Andrew Jarombek
 * @since 9/1/2022
 */

public class ThousandSeparator {
    public static String thousandSeparator(int n) {
        if (n == 0) {
            return "0";
        }

        StringBuilder sb = new StringBuilder();
        int i = 0;

        while (n > 0) {
            if (i > 0 && i % 3 == 0) {
                sb.append(".");
            }

            sb.append(n % 10);
            n /= 10;
            i++;
        }

        return sb.reverse().toString();
    }

    public static void main(String[] args) {
        assert thousandSeparator(123).equals("123");
        assert thousandSeparator(1234).equals("1.234");
    }
}
